observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java

```diff
@@ -158,18 +169,71 @@ class OpenTelemetrySubsystemDefinition extends PersistentResourceDefinition {
+                            Phase.DEPENDENCIES,
+                            0x1910,
+                            new OpenTelemetryDependencyProcessor());
+                    target.addDeploymentProcessor(OpenTelemetryConfigurationConstants.SUBSYSTEM_NAME,
+                            Phase.POST_MODULE,
+                            0x3810,
+                            new OpenTelemetryDeploymentProcessor(opentelemetryConfig));
+                })
+                .build();
+
+        ManagementResourceRegistrar.of(descriptor).register(registration);
 
-    protected OpenTelemetrySubsystemDefinition() {
-        super(new SimpleResourceDefinition.Parameters(OpenTelemetrySubsystemExtension.SUBSYSTEM_PATH,
-                OpenTelemetrySubsystemExtension.SUBSYSTEM_RESOLVER)
-                .setAddHandler(new OpenTelemetrySubsystemAdd())
-                .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE)
-                .setCapabilities(OPENTELEMETRY_CAPABILITY, OPENTELEMETRY_CONFIG_CAPABILITY)
+        return registration;
+    }
+
+    @Override
+    public ResourceServiceInstaller configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        String exporter = OpenTelemetrySubsystemRegistrar.EXPORTER.resolveModelAttribute(context, model).asString();
+        validateExporter(context, exporter);
+
+        final WildFlyOpenTelemetryConfig config = new WildFlyOpenTelemetryConfig(
+                OpenTelemetrySubsystemRegistrar.SERVICE_NAME.resolveModelAttribute(context, model).asStringOrNull(),
+                exporter,
+                OpenTelemetrySubsystemRegistrar.ENDPOINT.resolveModelAttribute(context, model).asStringOrNull(),
+                OpenTelemetrySubsystemRegistrar.BATCH_DELAY.resolveModelAttribute(context, model).asLongOrNull(),
+                OpenTelemetrySubsystemRegistrar.MAX_QUEUE_SIZE.resolveModelAttribute(context, model).asLongOrNull(),
+                OpenTelemetrySubsystemRegistrar.MAX_EXPORT_BATCH_SIZE.resolveModelAttribute(context, model).asLongOrNull(),
+                OpenTelemetrySubsystemRegistrar.EXPORT_TIMEOUT.resolveModelAttribute(context, model).asLongOrNull(),
+                OpenTelemetrySubsystemRegistrar.SPAN_PROCESSOR_TYPE.resolveModelAttribute(context, model).asStringOrNull(),
+                OpenTelemetrySubsystemRegistrar.SAMPLER.resolveModelAttribute(context, model).asStringOrNull(),
+                OpenTelemetrySubsystemRegistrar.RATIO.resolveModelAttribute(context, model).asDoubleOrNull(),
+                context.getCapabilityServiceSupport().hasCapability("org.wildfly.extension.microprofile.telemetry")
         );
+
+        Supplier<WildFlyOpenTelemetryConfig> configSupplier = () -> config;
+        opentelemetryConfig.set(config);
+
+        return CapabilityServiceInstaller.builder(OPENTELEMETRY_CONFIG_CAPABILITY, configSupplier)
+                .asActive()
+                .build();
```

Comment:
As this is a fixed value (i.e. resolvable the need for dependencies only available after starting), you can simplify this as:

```java
return CapabilityServiceInstaller.builder(OPENTELEMETRY_CONFIG_CAPABILITY, config).build();
```

Though, does this even need to be a service?  Where is it comsumed?

Reply:
The AtomicReference is consumed here: https://github.com/wildfly/wildfly/blob/c6370a7737e14a1d7cb4344a2b897cfe6d6c476a/observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java#L203

It seems weird to build a service (?) that offers that, but it's not consumed, while the value it offers is accessed via the AtomReference instead. I don't know how to do that more correctly, though, so advice is welcome here.

Reply:
In that case, you do not need to install a service.

Reply:
IIRC, I added that method so that I could build the WildFlyOpenTelemetryConfig instance, for which I need the context and model. To do that in configure(), I have to return a ResourceServiceInstaller. Is there a better way/place to build that WildFlyOpenTelemetryConfig? It doesn't look like I can do that in the DUP, which is the only place that consumes it, ultimately.

Reply:
> To do that in configure(), I have to return a ResourceServiceInstaller.

If your runtime does not install services you do not need ResourceServiceConfigurator.

Reply:
Where else would I build `WildFlyOpenTelemetryConfig`?